src/Tribe/Events/Attendee_List_Display.php

```diff
@@ -0,0 +1,223 @@
+<?php
+/**
+ * Class that detects if given post
+ * is displaying the Attendee List.
+ */
+
+namespace Tribe\Tickets\Events;
+
+use Exception;
+use InvalidArgumentException;
+use Tribe__Tickets__Main;
+use Tribe__Tickets_Plus__Attendees_List;
+use WP_Post;
+
+/**
+ * Class Attendee_List_Display
+ *
+ * @since   4.12.0
+ *
+ * @package Tribe\Tickets\Events
+ */
+class Attendee_List_Display {
+
+	/**
+	 * @var string Meta name to control whether the Attendee List
+	 *             meta was changed by a shortcode in the content.
+	 */
+	protected static $attendee_list_by_shortcode = 'tribe_tickets_attendee_list_triggered_by_shortcode';
+
+	/**
+	 * Check if given event is hiding the attendees list.
+	 *
+	 * @param int|WP_Post $post The Post being checked.
+	 * @throws InvalidArgumentException Could not determine if given event is hiding or showing the Attendee List.
+	 *
+	 * @return bool True if event is hiding the attendees list. False otherwise.
+	 */
+	public function is_event_hiding_attendee_list( $post ) {
+		// Attendees list is a Plus feature, if ET Plus is not present it will always be hidden.
+		if ( ! class_exists( 'Tribe__Tickets_Plus__Main' ) ) {
+			return true;
+		}
+
+		if ( is_numeric( $post ) ) {
+			$post = get_post( $post );
+		}
+
+		if ( ! $post instanceof WP_Post ) {
+			/*
+			 * There's no way to apply a sensible default in this scenario.
+			 *
+			 * Let's throw an exception so the client calling this function deals
+			 * with it according to the >> caller << context.
+			 */
+			throw new InvalidArgumentException( '$post should be either a post ID or a WP_Post object.' );
+		}
+
+		return (bool) Tribe__Tickets_Plus__Attendees_List::is_hidden_on( $post );
+	}
+
+	/**
+	 * @param bool $should_hide Whether the optout form should be hidden or not.
```

Comment:
```suggestion
	 * Determine whether we should hide the optout option.
	 *
	 * @since TBD
	 *
	 * @param bool $should_hide Whether the optout form should be hidden or not.
```